File: FinBoard/BackEnd/src/main/java/pwork/greco/antonio/finboard/entity/Instrument.java
package pwork.greco.antonio.finboard.entity;

import jakarta.persistence.*;
import lombok.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "Instrument")
public class Instrument {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(length = 100, nullable = false)
    private String name;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "InstrumentTypeId", nullable = false)
    private InstrumentType instrumentType;

    @Column(length = 10, nullable = false)
    private String currency;

    @Column(length = 20, unique = true)
    private String isin;
}

File: FinBoard/FrontEnd/src/app/models/Instrument.ts
export interface Instrument {
  id: number;                    // bigint
  currency: string;             // varchar(10)
  isin?: string | null;         // varchar(20) nullable
  name: string;                 // varchar(100)
  instrumentTypeId: number;     // bigint (FK a instrument_type)
  instrumentTypeDescription: string;     // bigint (FK a instrument_type)
}
